graph_weather/models/aurora/integration_layer.py

```diff
@@ -0,0 +1,256 @@
+import torch
+import torch.nn as nn
+import numpy as np
+from einops import rearrange
+from typing import Dict, Union, Tuple, Optional, List
+from dataclasses import dataclass
+from enum import Enum
+from torch_geometric.nn import knn
+from torch_geometric.utils import scatter
+from ..gencast.utils.noise import generate_isotropic_noise
+
+# Import Fengwu_GHR specific components
+from graph_weather.models.fengwu_ghr import (
+    ImageMetaModel,
+    WrapperImageModel,
+)
+
+class ModelType(Enum):
+    FENGWU_GHR = "fengwu_ghr"
+    GENCAST = "gencast"
+
+@dataclass
+class GenCastConfig:
+    """Configuration for GenCast model."""
+    hidden_dims: List[int] = None
+    num_blocks: int = 3
+    num_heads: int = 4
+    splits: int = 0
+    num_hops: int = 1
+
+@dataclass
+class Fengwu_GHRConfig:
+    """Configuration for Fengwu_GHR model."""
+    image_size: Union[int, Tuple[int, int]]
+    patch_size: Union[int, Tuple[int, int]]
+    depth: int
+    heads: int
+    mlp_dim: int
+    channels: int
+    dim_head: int = 64
+    scale_factor: Optional[Union[int, Tuple[int, int]]] = None
+
+class TransformationError(Exception):
+    """Custom exception for transformation errors."""
+    pass
+
+class ValidationError(Exception):
+    """Custom exception for validation errors."""
+    pass
+
+class IntegrationLayer:
```

Comment:
Ah, I see now, I think I might have misled you in the issue, what I meant is that the Aurora model implementation should be able to take a list of lat/lon points and features and run the model off of that, and then output back to a graph, like in this test: https://github.com/openclimatefix/graph_weather/blob/5908a56a3f5c1d08e62e4aa37f2b1bf05f72746f/tests/test_model.py#L34-L47 Not that it has to wrap GenCast or Fengwu-GHR. Sorry for the confusion. This should make this PR simpler though as you can remove these GenCast and Fengwu-GHR components.